server/src/maps/mod.rs

```diff
@@ -0,0 +1,305 @@
+use std::io::Cursor;
+
+use actix_web::{get, web, HttpRequest, HttpResponse};
+use awc::Client;
+use cached::lazy_static::lazy_static;
+use cached::proc_macro::cached;
+use cached::SizedCache;
+use futures::future::join_all;
+use image::Rgba;
+use imageproc::definitions::HasBlack;
+use imageproc::drawing::{draw_text_mut, text_size};
+use log::{debug, error, info};
+use rusqlite::{Connection, Error, OpenFlags};
+use rusttype::{Font, Scale};
+use tokio::time::Instant;
+
+use crate::utils;
+
+pub fn configure(cfg: &mut web::ServiceConfig) {
+    cfg.service(maps_handler);
+}
+
+struct MapInfo {
+    name: String,
+    type_common_name: String,
+    _type: String,
+    lat: f64,
+    lon: f64,
+}
+
+fn get_localised_data(id: &str, should_use_english: bool) -> Option<Result<MapInfo, Error>> {
+    let conn = Connection::open_with_flags(
+        "data/api_data.db",
+        OpenFlags::SQLITE_OPEN_READ_ONLY | OpenFlags::SQLITE_OPEN_NO_MUTEX,
+    )
+    .expect("Cannot open database");
+
+    let stmt =
+        match should_use_english {
+            false => conn
+                .prepare_cached("SELECT name,type,type_common_name,lat,lon FROM de WHERE key = ?"),
+            true => conn
+                .prepare_cached("SELECT name,type,type_common_name,lat,lon FROM en WHERE key = ?"),
+        };
```

Comment:
```suggestion
    let stmt = conn.prepare_cached(&format!(
        "SELECT name,type,type_common_name,lat,lon FROM {} WHERE key = ?",
        if should_use_english { "en" } else { "de" }
    ));
```
I would prefer it more like this, but it's not too important.